pandas/core/dtypes/concat.py

```diff
@@ -175,8 +175,8 @@ def is_nonempty(x):
         return _concat_sparse(to_concat, axis=axis, typs=typs)
 
     extensions = [is_extension_array_dtype(x) for x in to_concat]
-    if any(extensions):
-        to_concat = [np.atleast_2d(x.astype('object')) for x in to_concat]
+    if any(extensions) and axis == 1:
```

Comment:
so what happens on axis=0? coerced to``object``?

Reply:
Right (everything is upcast).